compose/cli/docker_client.py

```diff
@@ -16,19 +18,31 @@ def docker_client():
     base_url = os.environ.get('DOCKER_HOST')
     tls_config = None
 
-    if os.environ.get('DOCKER_TLS_VERIFY', '') != '':
+    tls = tls in ("true", "t", "1")
+    tls_verify = os.environ.get('DOCKER_TLS_VERIFY', '') != '' if tls_verify is None else tls_verify in ("true", "t", "1")
+    if tls or tls_verify:
         parts = base_url.split('://', 1)
         base_url = '%s://%s' % ('https', parts[1])
 
-        client_cert = (os.path.join(cert_path, 'cert.pem'), os.path.join(cert_path, 'key.pem'))
-        ca_cert = os.path.join(cert_path, 'ca.pem')
+        # Prefer cli argument over environment variable
+        tls_ca_cert = os.path.expanduser(os.path.join(cert_path, 'ca.pem') if tls_ca_cert is None else tls_ca_cert)
+        tls_cert = os.path.expanduser(os.path.join(cert_path, 'cert.pem') if tls_cert is None else tls_cert)
+        tls_key = os.path.expanduser(os.path.join(cert_path, 'key.pem') if tls_key is None else tls_key)
 
-        tls_config = tls.TLSConfig(
+        if not tls_verify:
+            warnings.warn((
+                'Unverified HTTPS request is being made. '
+                'Adding certificate verification is strongly advised. See: '
+                'https://urllib3.readthedocs.org/en/latest/security.html'),
+                errors.InsecureRequestWarning)
+            tls_ca_cert = None
+
+        tls_config = docker_tls.TLSConfig(
             ssl_version=ssl.PROTOCOL_TLSv1,
-            verify=True,
+            verify=tls_verify,
             assert_hostname=False,
-            client_cert=client_cert,
-            ca_cert=ca_cert,
+            client_cert=(tls_cert, tls_key),
+            ca_cert=tls_ca_cert,
```

Comment:
See https://github.com/docker/compose/pull/1744 - we want to use docker-py's `kwargs_from_env` rather than doing all this logic ourselves. That might well mean making changes to docker-py.